src/Exception/BatchException.php

```diff
@@ -2,61 +2,136 @@
 
 namespace Http\Client\Exception;
 
+use Http\Client\BatchResult;
+use Http\Client\Exception;
+use Psr\Http\Message\RequestInterface;
+
 /**
  * @author Márk Sági-Kazár <mark.sagikazar@gmail.com>
  */
-final class BatchException extends TransferException
+final class BatchException extends RuntimeException
 {
     /**
-     * @var TransferException[]
+     * @var BatchResult
+     */
+    private $result;
+
+    /**
+     * @var \SplObjectStorage
      */
     private $exceptions;
 
+    public function __construct()
+    {
+        $this->exceptions = new \SplObjectStorage();
+    }
+
     /**
-     * @param TransferException[] $exceptions
+     * Returns a list BatchResult which contains succesful responses
+     *
+     * @return BatchResult
      */
-    public function __construct(array $exceptions = [])
+    public function getResult()
     {
-        parent::__construct('An error occurred when sending multiple requests.');
+        return $this->result;
+    }
 
-        foreach ($exceptions as $e) {
-            if (!$e instanceof TransferException) {
-                throw new InvalidArgumentException('Exception is not an instanceof Http\Client\Exception\TransferException');
-            }
-        }
+    /**
+     * Sets the successful response list
+     *
+     * @param BatchResult $result
+     *
+     * @internal
+     */
+    public function setResult(BatchResult $result)
+    {
+        $this->result = $result;
```

Comment:
It is possible. But every `getResult` when no result is set will have to create a new object. It would be easier if a new object is created once and then it is set as the actual result.